lib/populator.js

```diff
@@ -267,12 +276,68 @@ Populator.prototype._enrollInProgram = function (knownKeys, trackedEntityInstanc
   }.bind(this));
 };
 
-Populator.prototype._checkForDuplicateEvent = function (knownKeys, trackedEntityInstanceID, duplicate, next) {
+Populator.prototype._checkForDuplicateEventUsingDataElementID = function (dataElements, trackedEntityInstanceID, duplicate, next) {
+  if (typeof duplicate === 'function') {
+    next = duplicate;
+    duplicate = false;
+  }
+  var uniqueEventDataElementID = this._cache.uniqueEventDataElementID
+  if (!dataElements[uniqueEventDataElementID]) {
+    return next(
+      new Error('No Data Element with ID: ' + uniqueEventDataElementID)
+    )
+  }
+
+  var requestTime = Date.now();
+
+  var request = Request.get({
+    url: URL.resolve(this._options.url, createDhisUrl(this._options.dhisVersion, 'events')),
+    qs: {
+      trackedEntityInstance: trackedEntityInstanceID,
+      fields: 'dataValues[dataElement,value]'
```

Comment:
Should we not maybe consider adding in a `pageSize` on this events query. This would need to be quite a high number. The default is 50 events and if there are more than 50 we will not find the duplicate event.

I wonder if its possible to do a event lookup based on a specific data element value?